sqlglot/dialects/snowflake.py

```diff
@@ -393,6 +393,8 @@ class Parser(parser.Parser):
             ),
             "BITXOR": binary_from_function(exp.BitwiseXor),
             "BIT_XOR": binary_from_function(exp.BitwiseXor),
+            "BITOR": binary_from_function(exp.BitwiseOr),
+            "BIT_OR": binary_from_function(exp.BitwiseOr),
```

Comment:
Do note that `BITOR` has a 3rd argument as well:

```
BITOR( <expr1> , <expr2> [ , '<padside>' ] )
```

Thus `binary_from_function` here seems problematic. The easy solution would be to fallback to an Anonymous function in that case, i.e you'd need to create a builder function that will return:
- `binary_from_function(...)` if `args == 2` or
- `exp.Anonymous(...)` if `args == 3`.

For instance, check out `snowflake.py::_build_datetime()` or `bigquery.py::_build_contains_substring()`. 

We should also test that the 3-arg version roundtrips fine in this PR too.

Reply:
Hi @VaggelisD 
I have added the suggested changes. Will you please check it in your free time?

Reply:
I believe we generally have these functions / binary operators inherit from both `Binary` and `Func`. That takes care of the argument parsing. Let's keep what's already in this PR though and we can revisit if/when needed.